cpp/src/arrow/python/datetime.cc

```diff
@@ -450,6 +475,19 @@ Result<std::string> TzinfoToString(PyObject* tzinfo) {
   return PyTZInfo_utcoffset_hhmm(tzinfo);
 }
 
+Result<PyObject*> MonthDayNanoIntervalToNamedTuple(
```

Comment:
Since NULL is return an error, you don't need to wrap the return value in a `Result<>`.

Reply:
done.

Reply:
There are a few other places that return Result<PyObject*> is there any guidance on when to use that and when to use pure PyObject*?